folio-spring-i18n/src/main/java/org/folio/spring/i18n/model/TranslationFile.java

```diff
@@ -0,0 +1,148 @@
+package org.folio.spring.i18n.model;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import lombok.With;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.core.io.Resource;
+
+/**
+ * A class to represent a file that contains translation information.  This file will have a name
+ * identifying its associated locale with its contents being an ICU dictionary.
+ *
+ * <p>This class also contains several utility methods for parsing filenames and aggregating
+ * {@code TranslationFile} objects.</p>
+ */
+@Value
+@Log4j2
+@AllArgsConstructor
+public class TranslationFile {
+
+  /**
+   * The String denoting that a language or country is unknown from {@link getParts}.
+   */
+  public static final String UNKNOWN_PART = "*";
+
+  private static final int MAX_FILENAME_PARTS = 2;
+  private static final String JSON_FILE_SUFFIX = ".json";
+
+  /**
+   * The resources backing the translation files.
+   */
+  private List<Resource> resources;
+
+  /**
+   * Get the map of patterns from this file.
+   *
+   * @return the map of patterns -&gt; ICU format strings
+   */
+  public Map<String, String> getPatterns() {
+    Map<String, String> result = new HashMap<>();
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    for (Resource resource : resources) {
+      try {
+        Map<String, String> moduleMap = objectMapper.readValue(
+          resource.getInputStream(),
+          new TypeReference<Map<String, String>>() {}
+        );
+
+        String moduleName = Path
+          .of(resource.getURL().getPath())
+          .getParent()
+          .getFileName()
+          .toString();
+
+        moduleMap.forEach((key, value) ->
+          result.put(moduleName + "." + key, value)
+        );
+      } catch (IOException e) {
+        log.error(
+          "Could not open/read translation file {}; will use fallback instead",
+          resource.getDescription(),
+          e
+        );
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Get the parts of this filename as a {@link LanguageRegionPair LanguageRegionPair}.
+   *
+   * <p>If one (or either) is unknown, {@code UNKNOWN_PART} will be returned.</p>
+   *
+   * @return a {@link LanguageRegionPair LanguageRegionPair}
+   */
+  public LanguageRegionPair getParts() {
```

Comment:
Could we rename it to something like getLanguageRegion? I think it will be more clear.